app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -140,6 +141,16 @@ export const reducer = handleActions({
 
   [LISTEN_SIGNALS_PUT_REQUEST_FAIL]: handlesListenSignalFailResponse,
 
+  [PROMOTION_SIGNAL_POST_REQUEST_SUCCESS]: (state, action) => {
+    const promotionalTweet = _.get(action.payload, 'promotional_tweet', {});
+    const id = promotionalTweet.id;
+    const signalId = promotionalTweet.listen_signal_id
+    const data = state.data;
+
+    // Can't figure this out here....??
```

Comment:
This should just be:

```jsx
return {
  ..state,
  data: {
    ...state.data,
    [id]: promotionalTweet,
  },
};
```

Since this isn't a reversible action (`POST`), you shouldn't have to worry about the previous state of the promotional tweet; this should only be additive.